server/controllers/userController.js

```diff
@@ -28,9 +25,12 @@ userController.createUser = (req, res, next) => {
         key
       ];
 
-      db.query(query, values);
+      return db.query(query, values);
     })
     .then(response => {
+      if (!response.rows.length) {
+        throw new Error('Username already taken');
```

Comment:
Should we throw an error or invoke the error handler instead?

Reply:
fixed. will be included in next commit.